lib/quantum/executor.ex

```diff
@@ -159,15 +159,6 @@ defmodule Quantum.Executor do
   def log_exception(kind, reason, stacktrace) do
     reason = Exception.normalize(kind, reason, stacktrace)
 
-    crash_reason =
-      case kind do
-        :throw -> {{:nocatch, reason}, stacktrace}
-        _ -> {reason, stacktrace}
-      end
-
-    Logger.error(
-      Exception.format(kind, reason, stacktrace),
-      crash_reason: crash_reason
-    )
+    Logger.error(Exception.format(kind, reason, stacktrace))
```

Comment:
The crash reason metadata is actually quite important. Have a look at this PR: https://github.com/quantum-elixir/quantum-core/pull/464

We also discussed it together with Jose Va im via the mailing list.

Reply:
According to the docs (https://hexdocs.pm/logger/Logger.html) `:crash_reason` **is always added automatically by Logger whenever possible**

Reply:
@pnezis Except we're logging manually and it can't do it automatically.

https://groups.google.com/g/elixir-lang-core/c/pWz-uTVMEVM/m/QGXncxD1AQAJ

Reply:
```elixir
defmodule CaptureLogErrorTest do
  use ExUnit.Case

  require Logger

  import ExUnit.CaptureLog

  test "greets the world" do
    assert capture_log(fn ->
             try do
               raise "foo"
             rescue
               error in RuntimeError ->
                 reason = Exception.normalize(:error, error, __STACKTRACE__)

                 Logger.error(
                   Exception.format(:error, reason, __STACKTRACE__),
                   crash_reason: {reason, __STACKTRACE__}
                 )
             end
           end) =~ "foo"
  end
end
```

This produces no error for me. Does it for you?
```

Reply:
Oh this only happens in Elixir < 1.10.

Then I'd rather just raise the elixir requirement.

Reply:
Or detect the elixir version and only do it in Elixir `~> 1.10` with a `TODO` to remove the condition in the future.

Reply:
I confirm it is an elixir 1.9 issue. I will add the version check and push

Reply:
@maennchen Added the version check.